NOnion/Utility/Base32Util.fs

```diff
@@ -0,0 +1,171 @@
+
+            let quintuplet =
+                (int64 x0 <<< 32)
+                ||| (int64 x1 <<< 24)
+                ||| (int64 x2 <<< 16)
+                ||| (int64 x3 <<< 8)
+                ||| (int64 x4)
+
+            let a0 = (quintuplet &&& 0xF800000000L) >>> 35 |> int
+            let a1 = (quintuplet &&& 0x07C0000000L) >>> 30 |> int
+            let a2 = (quintuplet &&& 0x003E000000L) >>> 25 |> int
+            let a3 = (quintuplet &&& 0x0001F00000L) >>> 20 |> int
+            let a4 = (quintuplet &&& 0x00000F8000L) >>> 15 |> int
+            let a5 = (quintuplet &&& 0x0000007C00L) >>> 10 |> int
+            let a6 = (quintuplet &&& 0x00000003E0L) >>> 5 |> int
+            let a7 = (quintuplet &&& 0x000000001FL) |> int
+
+            match ending with
+            | 1 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    '='
+                    '='
+                    '='
+                    '='
+                    '='
+                    '='
+                ] // 01======
+            | 2 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    '='
+                    '='
+                    '='
+                    '='
+                ] // 0123====
+            | 3 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    alphabet.[a4]
+                    '='
+                    '='
+                    '='
+                ] // 01234===
+            | 4 ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    alphabet.[a4]
+                    alphabet.[a5]
+                    alphabet.[a6]
+                    '='
+                ] // 0123456=
+            | _ ->
+                [
+                    alphabet.[a0]
+                    alphabet.[a1]
+                    alphabet.[a2]
+                    alphabet.[a3]
+                    alphabet.[a4]
+                    alphabet.[a5]
+                    alphabet.[a6]
+                    alphabet.[a7]
+                ] // 01234567
+
+        let rec parse result input =
+            match input with
+            | x0 :: x1 :: x2 :: x3 :: x4 :: tail ->
+                parse (result @ quintupletToList 5 (x0, x1, x2, x3, x4)) tail
+            | x0 :: x1 :: x2 :: [ x3 ] ->
+                result @ quintupletToList 4 (x0, x1, x2, x3, 0uy)
+            | x0 :: x1 :: [ x2 ] ->
+                result @ quintupletToList 3 (x0, x1, x2, 0uy, 0uy)
+            | x0 :: [ x1 ] ->
+                result @ quintupletToList 2 (x0, x1, 0uy, 0uy, 0uy)
+            | [ x0 ] -> result @ quintupletToList 1 (x0, 0uy, 0uy, 0uy, 0uy)
```

Comment:
let's use `foo :: []` instead of `[foo]`, in the 4 lines above

Reply:
FSharpLint made me change this

Reply:
what? let's disable this rule, which one is it?

Reply:
ah it's a hint, let's inverse it, instead of "x::[] ===> [x]", make it "[x] ===> x::[]",

Reply:
Fixed in ce823ce0ac6f58340c96927ffea32d8b5a6d628dv.